File: app-java/src/main/java/com/tritiumstudios/phasmophobiaevidencetool/android/activities/mainmenus/marketplace/views/MarketplaceListLayout.java
package com.tritiumstudios.phasmophobiaevidencetool.android.activities.mainmenus.marketplace.views;

import android.animation.LayoutTransition;
import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatTextView;

import com.tritiumstudios.phasmophobiaevidencetool.R;

public class MarketplaceListLayout extends LinearLayout {

    public MarketplaceListLayout(Context context) {
        super(context);

        init();
    }

    public MarketplaceListLayout(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);

        init();
    }

    public MarketplaceListLayout(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        init();
    }

    public MarketplaceListLayout(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);

        init();
    }

    public void init() {
        inflate(getContext(), R.layout.layout_marketplace_grouplabel, this);

        setLayoutTransition(new LayoutTransition());

        setOrientation(LinearLayout.VERTICAL);
    }

    public void setOrientation(int orientation) {
        super.setOrientation(orientation);
        new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
    }

    public void setLabel(String text) {
        AppCompatTextView label = findViewById(R.id.label);
        if(label != null) {
            label.setText(text);
        }
    }

    public void showLabel(int visibility) {
        AppCompatTextView label = findViewById(R.id.label);
        if(label != null) {
            label.setVisibility(visibility);
        }
    }

    public void validateItems() {
        for(int i = 0; i < getChildCount(); i++) {
            View child = getChildAt(i);
            if(child instanceof ThemeSingleCardView singleThemeView) {
                singleThemeView.validate();
            }
            else if(child instanceof ThemeBundleCardView bundleThemeView) {
                bundleThemeView.validate();
            }
        }
    }
}

File: app/src/main/java/com/tritiumgaming/phasmophobiaevidencepicker/activities/mainmenus/marketplace/views/MarketplaceListLayout.kt
package com.tritiumgaming.phasmophobiaevidencepicker.ui.mainmenus.marketplace.views

import android.animation.LayoutTransition
import android.content.Context
import android.util.AttributeSet
import android.widget.LinearLayout
import androidx.appcompat.widget.AppCompatTextView
import com.tritiumgaming.phasmophobiaevidencepicker.R
import com.tritiumgaming.phasmophobiaevidencepicker.ui.mainmenus.marketplace.views.items.MarketBundleView
import com.tritiumgaming.phasmophobiaevidencepicker.ui.mainmenus.marketplace.views.items.MarketThemeView

class MarketplaceListLayout : LinearLayout {
    constructor(context: Context?) : super(context)

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) :
            super(context, attrs, defStyleAttr)

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) :
            super(context, attrs, defStyleAttr, defStyleRes)

    init {
        inflate(context, R.layout.layout_marketplace_list, this)

        layoutTransition = LayoutTransition()
        orientation = VERTICAL
    }

    fun setLabel(text: String?) {
        val label = findViewById<AppCompatTextView>(R.id.label)
        label?.text = text
    }

    fun showLabel(visibility: Int) {
        val label = findViewById<AppCompatTextView>(R.id.label)
        label?.visibility = visibility
    }

    fun validateChildren() {
        for (i in 0 until childCount) {
            when(val child = getChildAt(i)) {
                is MarketThemeView -> child.validate()
                is MarketBundleView -> child.validate()
            }
        }
    }

    override fun setOrientation(orientation: Int) {
        super.setOrientation(orientation)
        layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT)
    }

}
